### PR TITLE
model: add nemotron rerank

### PR DESCRIPTION
**Pull request by Samoed**
Tried to run on `AskUbuntuDupQuestions` and got `0.63817`

Close https://github.com/embeddings-benchmark/mteb/issues/3720

### CODE REVIEW

File: mteb/models/sentence_transformer_wrapper.py [hunk 1]
```diff
@@ -261,6 +261,8 @@ def __init__(
         self,
         model: CrossEncoder | str,
         revision: str | None = None,
+        query_prefix: str | None = None,
+        passage_prefix: str | None = None,
```
Comment (isaac-chung) [inline mteb/models/sentence_transformer_wrapper.py]: How about directly setting empty string as default?
```suggestion
        query_prefix: str = "",
        passage_prefix: str = "",
```

File: mteb/models/model_implementations/nvidia_llama_nemoretriever_colemb.py [hunk 2]
```diff
@@ -34,6 +32,14 @@ def __init__(
         attn_implementation="flash_attention_2",
         **kwargs,
     ):
+        required_transformers_version = "4.49.0"
+
+        if Version(transformers_version) != Version(required_transformers_version):
+            raise RuntimeError(
+                f"transformers version {transformers_version} is not match with required "
+                f"install version {required_transformers_version} to run `nvidia/NV-Embed-v2`"
```
Comment (isaac-chung) [inline mteb/models/model_implementations/nvidia_llama_nemoretriever_colemb.py]: Maybe it's this model based on the filename?
```suggestion
                f"install version {required_transformers_version} to run `nvidia/llama-nemoretriever-colembed`"
```

File: mteb/models/model_implementations/nvidia_models.py [hunk 3]
```diff
@@ -547,3 +548,55 @@ def _extract_embeddings(
     contacts=["ybabakhin"],
     citation=NV_RETRIEVER_CITATION,
 )
+
+
+def _nemotron_rerank_model(model: str, revision: str, **kwargs) -> CrossEncoderWrapper:
```
Comment (isaac-chung) [inline mteb/models/model_implementations/nvidia_models.py:635]: Where is this function used?

**Review by isaac-chung — COMMENTED** [review]
There is an evaluation results section from the model card. Maybe compare one result from there?

File: mteb/models/model_implementations/nvidia_models.py (hunk 3 above)
Comment (Samoed) [reply mteb/models/model_implementations/nvidia_models.py:635]: This is loader function for `nvidia/llama-nemotron-rerank-1b-v2`

https://github.com/embeddings-benchmark/mteb/blob/01e6334a04dc5eb474f37e56b3fc80d54f3d8be3/mteb/models/model_implementations/nvidia_models.py#L569-L571

**Comment by Samoed** [comment]
@ybabakhin @jzakrzew Can you review implementation of the model?

> There is an evaluation results section from the model card. Maybe compare one result from there?

They have evaluated only as 2 stage pipeline. I can do similarly, but later because this requires changes in my venv

**Comment by ybabakhin** [comment]
Thank you @Samoed! I will take a look later this week

File: mteb/models/model_implementations/nvidia_models.py [hunk 4]
```diff
@@ -550,3 +551,55 @@ def _extract_embeddings(
...
+    required_transformers_version = "4.47.1"
+
+    if Version(transformers_version) != Version(required_transformers_version):
+        raise RuntimeError(
+            f"transformers version {transformers_version} is not match with required "
+            f"install version {required_transformers_version} to run `nvidia/NV-Embed-v2`"
```
Comment (ybabakhin) [inline mteb/models/model_implementations/nvidia_models.py]:
```suggestion
            f"install version {required_transformers_version} to run `nvidia/llama-nemotron-rerank-1b-v2`"
```

File: mteb/models/model_implementations/nvidia_models.py [hunk 5]
```diff
@@ -550,3 +551,55 @@ def _extract_embeddings(
...
+    )
+
+
+nemotron_rerank_1b_v2 = ModelMeta(
+    loader=_nemotron_rerank_model,
+    loader_kwargs=dict(
+        trust_remote_code=True,
+        query_prefix="question:",
+        passage_prefix=" \n \n passage:",
+        model_kwargs={"torch_dtype": torch.bfloat16},
```
Comment (ybabakhin) [inline mteb/models/model_implementations/nvidia_models.py]: I would either change dtype to `float32` here, or add a separate item to `loader_kwargs` -- `activation_fn=torch.nn.Identity()`. By default, SentenceTransformers apply a sigmoid activation which in a combination with bfloat16 loses a lot information.

cc: @tomaarsen 

I did a run on FIQA, here are the metrics and one sample result for each setup:
* `Sigmoid + bf16` (current implementation). Score: 0.31996
```
 '376148': 1.0,
 '580025': 1.0,
 '497993': 0.99609375,
 '253614': 0.9296875,
 '32833': 0.002471923828125,
```

* `Sigmoid + fp32`. Score: 0.32357
```
 '376148': 0.9999889135360718,
 '580025': 0.9995622038841248,
 '497993': 0.9968037605285645,
 '253614': 0.9362275004386902,
 '32833': 0.002313266508281231,
```

* `Identity + bf16`. Score: 0.32423
```
 '376148': 11.375,
 '580025': 7.78125,
 '497993': 5.6875,
 '253614': 2.609375,
 '32833': -6.0,
```

* `Identity + fp32`. Score: 0.32414
```
 '376148': 11.409661293029785,
 '580025': 7.733245849609375,
 '497993': 5.742588520050049,
 '253614': 2.6865363121032715,
 '32833': -6.066778659820557,
```

Comment (Samoed) [reply mteb/models/model_implementations/nvidia_models.py]: Changed to fp32. I just not sure that passing `torch.nn.Identity` is good approach, because this wouldn't be saved in config on cross-encoder

Comment (tomaarsen) [reply mteb/models/model_implementations/nvidia_models.py]: Perhaps the model config could also be updated if you prefer Identity? Then it'll always be loaded with that. 

I'm also open to changing the default activation function on Sentence Transformers, as that `bf16 + Sigmoid` issue seems like it would very commonly prevent distinguishing solid documents (e.g. twice `1.0` instead of `11.375` & `7.78125`). I might have to wait until the next major release for that though.

Comment (ybabakhin) [reply mteb/models/model_implementations/nvidia_models.py]: `fp32` is fine for this PR.

> Perhaps the model config could also be updated if you prefer Identity?

Will look into this

**Review by ybabakhin — APPROVED** [review]
Thank you @Samoed , LGTM!
